README.md
# mx-boot-options
Program for controlling boot options in MX Linux

[![build result](https://build.opensuse.org/projects/home:mx-packaging/packages/mx-boot-options/badge.svg?type=default)](https://software.opensuse.org//download.html?project=home%3Amx-packaging&package=mx-boot-options)


debian/changelog
mx-boot-options (23.11.05) mx; urgency=medium

  * Fix: use full path for plymouth-set-theme

 -- Adrian <adrian@mxlinux.org>  Wed, 15 Nov 2023 16:29:46 -0500

mx-boot-options (23.11.04) mx; urgency=medium

  * Fix listing Grub entries, it does require elevation unfortunately

 -- Adrian <adrian@mxlinux.org>  Sun, 12 Nov 2023 21:17:03 -0500

mx-boot-options (23.11.03) mx; urgency=medium

  * Don't prompt for admin password when loading the GUI if not running as chroot

 -- Adrian <adrian@mxlinux.org>  Sun, 12 Nov 2023 11:51:35 -0500

mx-boot-options (23.11.02) mx; urgency=medium

  * Run program as normal user and elevate only when running specific commands
  * Display boot log in a native window instead of terminal
  * Minor fixes

 -- Adrian <adrian@mxlinux.org>  Sun, 05 Nov 2023 01:11:09 -0400

mx-boot-options (23.8.01) mx; urgency=medium

  * Fix applying different message levels

 -- Adrian <adrian@mxlinux.org>  Thu, 10 Aug 2023 13:12:39 -0400

mx-boot-options (23.8) mx; urgency=medium

  * Fix displaying bootlog

 -- Adrian <adrian@mxlinux.org>  Thu, 10 Aug 2023 00:34:18 -0400

mx-boot-options (23.7.01) mx; urgency=medium

  * Update translations

 -- Adrian <adrian@mxlinux.org>  Sun, 23 Jul 2023 11:15:55 -0400

mx-boot-options (23.7) mx; urgency=medium

  * Update translations

 -- Adrian <adrian@mxlinux.org>  Thu, 20 Jul 2023 16:05:36 -0400

mx-boot-options (23.6) mx; urgency=medium

  * Update policy file
  * Refresh translations

 -- Adrian <adrian@mxlinux.org>  Fri, 09 Jun 2023 00:22:39 -0400

mx-boot-options (23.5) mx; urgency=medium

  * Update logo

 -- Adrian <adrian@mxlinux.org>  Tue, 16 May 2023 21:24:18 -0400

mx-boot-options (23.4) mx; urgency=medium

  * Update icon
  * Update translations

 -- Adrian <adrian@mxlinux.org>  Thu, 13 Apr 2023 22:42:32 -0400

mx-boot-options (23.2.01) mx; urgency=medium

  * Change ownership of the rc file created by KDE/Plasma

 -- Adrian <adrian@mxlinux.org>  Mon, 27 Feb 2023 19:57:54 -0500

mx-boot-options (23.2) mx; urgency=medium

  * Disable saving last boot choice on Btrs (not supported)
  * Update translations

 -- Adrian <adrian@mxlinux.org>  Sun, 19 Feb 2023 13:03:57 -0500

mx-boot-options (23.1) mx; urgency=medium

  * Fix: disable boot splash
  * Update translations

 -- Adrian <adrian@mxlinux.org>  Mon, 30 Jan 2023 20:15:16 -0500

mx-boot-options (22.10.02) mx; urgency=medium

  * Allow skipping selecting directory for chroot when running Live (useful for managing Uefi boot options)

 -- Adrian <adrian@mxlinux.org>  Thu, 27 Oct 2022 22:58:53 -0400

mx-boot-options (22.10.01) mx; urgency=medium

  * update translations 

 -- Dolphin Oracle <dolphinoracle@gmail.com>  Wed, 26 Oct 2022 19:56:18 -0400

mx-boot-options (22.9.01) mx; urgency=medium

  * Fix: toggling flatmenu. Make sure it can read "true" last entry wins

 -- fehlix <fehlix@mxlinux.org>  Thu, 15 Sep 2022 20:04:45 -0400

mx-boot-options (22.9) mx; urgency=medium

  * Fix: toggling flatmenu. Make sure it can read both "y" and "yes"

 -- Adrian <adrian@mxlinux.org>  Thu, 15 Sep 2022 19:37:17 -0400

mx-boot-options (22.8.01) mx; urgency=medium

  * Fix: launching documents from Help (fixes a problem in Plasma)

 -- Adrian <adrian@mxlinux.org>  Mon, 22 Aug 2022 21:02:26 -0400

mx-boot-options (22.8) mx; urgency=medium

  * Fix: launching Firefox from Help

 -- Adrian <adrian@mxlinux.org>  Wed, 17 Aug 2022 21:36:22 -0400

mx-boot-options (22.7.01) mx; urgency=medium

  * translation refresh

 -- Dolphin Oracle <dolphinoracle@gmail.com>  Sat, 23 Jul 2022 11:15:08 -0400

mx-boot-options (22.7) mx; urgency=medium

  * Adjust pkexec/sudo environment parameter
  * Fix EFI System Partition detection on dos/mbr partition table
  * Remove not needed --qtenv annotation pkexec policy

 -- fehlix <fehlix@mxlinux.org>  Mon, 04 Jul 2022 19:18:59 -0400

mx-boot-options (22.6.02) mx; urgency=medium

  * Fix: When creating UEFI entry use correct partition number rather than default #1

 -- Adrian <adrian@mxlinux.org>  Thu, 30 Jun 2022 22:08:19 -0400

mx-boot-options (22.6.01) mx; urgency=medium

  * Add UEFI boot manager

 -- Adrian <adrian@mxlinux.org>  Mon, 27 Jun 2022 03:07:23 -0400

mx-boot-options (22.6) mx; urgency=medium

  * Preserve user theme when launching program as root
  * Make sure HOME and XDG_RUNTIME_DIR are correct when launching as root
  * Advance animation when previewing plymouth theme

 -- Adrian <adrian@mxlinux.org>  Mon, 13 Jun 2022 13:50:52 -0400

mx-boot-options (22.2) mx; urgency=medium

  * New build

 -- Adrian <adrian@mxlinux.org>  Thu, 10 Feb 2022 08:59:27 -0500

mx-boot-options (21.1.02) mx; urgency=medium

  * adjust chroot

 -- fehlix <fehlix@mxlinux.org>  Thu, 10 Feb 2022 07:36:02 -0500

mx-boot-options (21.1.01) mx; urgency=medium

  * bug fix: detection grub present 


translations/mx-boot-options_af.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1" language="af">
<context>
    <name>CustomDialog</name>
    <message>
        <location filename="../dialog.cpp" line="16"/>
        <source>Live environment detected. Please select the root partition of the
 system you want to modify (only Linux partitions are displayed)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../dialog.cpp" line="22"/>
        <source>OK</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../dialog.cpp" line="23"/>
        <source>Cancel</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>MainWindow</name>
    <message>
        <location filename="../mainwindow.ui" line="33"/>
        <source>Options</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="39"/>
        <source>Use simplified menu structure without submenus</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="42"/>
        <source>Use flat menus (no submenus)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="55"/>
        <source>Boot to</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="78"/>
        <source>Set to &apos;0&apos; to boot immediately without displaying the menu, or to &apos;-1&apos; to wait indefinitely</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="81"/>
        <source>Menu timeout</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="104"/>
        <source>Kernel parameters</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="117"/>
        <source>seconds</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="124"/>
        <source>With this option enabled whatever entry you select from the grub boot menu will be saved as the new default for future boots</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="127"/>
        <source>Enable saving last boot choice</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="134"/>
        <source>Manage UEFI Boot Options</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="150"/>
        <source>Background</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="168"/>
        <source>Image</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="181"/>
        <location filename="../mainwindow.ui" line="283"/>
        <source>Enable theme</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="261"/>
        <source>Splash</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="296"/>
        <source>Preview</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="319"/>
        <source>Messages</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="325"/>
        <source>Very detailed</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="332"/>
        <source>Detailed (default setting)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="342"/>
        <source>Limited</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="358"/>
        <source>Display log</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="411"/>
        <source>Display help </source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="414"/>
        <source>Help</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="421"/>
        <source>Alt+H</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="453"/>
        <source>About this application</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="456"/>
        <source>About...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="463"/>
        <source>Alt+B</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="498"/>
        <source>Quit application</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="501"/>
        <source>Close</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="508"/>
        <source>Alt+N</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.ui" line="549"/>
        <source>Apply</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="89"/>
        <source>Still running</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="90"/>
        <source>Process still running. Are you sure you want to quit?</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="237"/>
        <source>Set timeout</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="237"/>
        <source>Timeout in seconds:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="242"/>
        <location filename="../mainwindow.cpp" line="482"/>
        <location filename="../mainwindow.cpp" line="1119"/>
        <source>Timeout: %1 seconds</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="255"/>
        <location filename="../mainwindow.cpp" line="484"/>
        <location filename="../mainwindow.cpp" line="1118"/>
        <location filename="../mainwindow.cpp" line="1134"/>
        <source>Boot Next: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="263"/>
        <source>Removal confirmation</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="264"/>
        <source>Are you sure you want to delete this boot entry?
%1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="347"/>
        <location filename="../mainwindow.cpp" line="350"/>
        <source>Select EFI file</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="348"/>
        <location filename="../mainwindow.cpp" line="351"/>
        <source>EFI files (*.efi *.EFI)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="361"/>
        <location filename="../mainwindow.cpp" line="375"/>
        <location filename="../mainwindow.cpp" line="406"/>
        <location filename="../mainwindow.cpp" line="541"/>
        <location filename="../mainwindow.cpp" line="612"/>
        <source>Error</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="362"/>
        <source>Could not find the source mountpoint for %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="366"/>
        <source>Set name</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="366"/>
        <source>Enter the name for the UEFI menu item:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="375"/>
        <source>Something went wrong, could not add entry.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="393"/>
        <source>Installing bootsplash, please wait</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="400"/>
        <source>Updating sources</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="402"/>
        <source>Installing</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="406"/>
        <source>Could not install the bootsplash.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="486"/>
        <source>Boot Current: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="541"/>
        <source>Could not create a temporary folder</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="551"/>
        <source>Cannot continue</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="552"/>
        <source>Cannot create chroot environment, cannot change boot options. Exiting...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="612"/>
        <source>Something went wrong, could not save boot order.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="818"/>
        <location filename="../mainwindow.cpp" line="1242"/>
        <source>Updating configuration, please wait</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="871"/>
        <source>Updating initramfs...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="883"/>
        <location filename="../mainwindow.cpp" line="1255"/>
        <source>Updating grub...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="887"/>
        <source>Done</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="887"/>
        <source>Changes have been successfully applied.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="899"/>
        <source>About %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="900"/>
        <source>Version: </source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="902"/>
        <source>Program for selecting common start-up choices</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="904"/>
        <source>Copyright (c) MX Linux</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="905"/>
        <source>%1 License</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="926"/>
        <source>%1 Help</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="935"/>
        <location filename="../mainwindow.cpp" line="1220"/>
        <source>Running in a Virtual Machine</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="936"/>
        <location filename="../mainwindow.cpp" line="1221"/>
        <source>You current system is running in a Virtual Machine,
Plymouth bootsplash will work in a limited way, you also won&apos;t be able to preview the theme</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="943"/>
        <source>Plymouth packages not installed</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="944"/>
        <source>Plymouth packages are not currently installed.
OK to go ahead and install them?</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="967"/>
        <source>Images (*.png *.jpg *.jpeg *.tga)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1079"/>
        <source>Boot log</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1093"/>
        <source>Log not found</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1093"/>
        <source>Could not find log at </source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1100"/>
        <source>Edit UEFI Boot Entries</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1103"/>
        <source>You can use the Up/Down buttons, or drag &amp; drop items to change boot order.
- Items are listed in the boot order.
- Grayed out lines are inactive.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1107"/>
        <location filename="../mainwindow.cpp" line="1161"/>
        <source>Set ac&amp;tive</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1108"/>
        <source>&amp;Add entry</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1109"/>
        <source>Boot &amp;next</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1084"/>
        <location filename="../mainwindow.cpp" line="1110"/>
        <source>&amp;Close</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1111"/>
        <source>Move &amp;down</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1112"/>
        <source>&amp;Remove entry</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1113"/>
        <source>Re&amp;set next</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1114"/>
        <source>Change &amp;timeout</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1115"/>
        <source>Move &amp;up</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1118"/>
        <location filename="../mainwindow.cpp" line="1134"/>
        <source>not set, will boot using list order</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1158"/>
        <source>Set &amp;inactive</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1211"/>
        <source>Needs reboot</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1212"/>
        <source>Plymouth was just installed, you might need to reboot before being able to display previews</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1276"/>
        <source>Click to select theme</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>QApplication</name>
    <message>
        <location filename="../main.cpp" line="57"/>
        <source>Program for selecting common start-up choices</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>QObject</name>
    <message>
        <location filename="../about.cpp" line="56"/>
        <source>License</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../about.cpp" line="57"/>
        <location filename="../about.cpp" line="66"/>
        <source>Changelog</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../about.cpp" line="58"/>
        <source>Cancel</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../about.cpp" line="77"/>
        <source>&amp;Close</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../main.cpp" line="85"/>
        <location filename="../main.cpp" line="94"/>
        <source>Error</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../main.cpp" line="86"/>
        <source>You seem to be logged in as root, please log out and log in as normal user to use this program.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../main.cpp" line="95"/>
        <source>You must run this program with admin access.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="964"/>
        <source>Select image to display in bootloader</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../mainwindow.cpp" line="1286"/>
        <source>Select GRUB theme</source>
        <translation type="unfinished"></translation>
    </message>
</context>
</TS>
